pkgs/native_toolchain_c/lib/src/cbuilder/clinker.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+
+import 'ctool.dart';
+import 'language.dart';
+import 'linker_options.dart';
+import 'linkmode.dart';
+import 'output_type.dart';
+import 'run_cbuilder.dart';
+
+export 'linker_options.dart';
+
+/// Specification for linking an artifact with a C linker.
```

Comment:
Please add a TODO that only Linux is supported and link it to an open GitHub issue.

(You could mark the issue open to contributions.)